pulsar: 1.109.0 -> 1.114.0

bryango opened the pull request:
## Description of changes

Release notes: https://github.com/pulsar-edit/pulsar/releases/tag/v1.114.0

A patchelf fixup is no longer valid in v1.114 so it is simply removed.

`pulsar` has been marked insecure so this change would not be verified by ofborg. Still, it is very actively developed and quite enjoyable to use. As an atom veteran, I would like to help keep it usable; thus adding myself to the maintainers.

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

bryango commented:
Result of `nixpkgs-review pr 292611 --extra-nixpkgs-config '{ allowInsecurePredicate = x: true; }'` run on `x86_64-linux`
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>pulsar</li>
  </ul>
</details>

COLAMAroro reviewed (approved):
I do not have Nix(OS) devices to actually test anymore, but LGTM.

Maybe consider running the AArch64 build using `binfmt` ?

bryango commented:
Thank you very much for the review!

> Maybe consider running the AArch64 build using `binfmt` ?

I will try to do this and see how it goes. Never used `binfmt` but looks promising!

COLAMAroro commented:
> I will try to do this and see how it goes. Never used binfmt but looks promising!

You'll be able to run the native AArch64 build using `nix build .#legacyPackages.aarch64-linux.pulsar` through binfmt/qemu.

Do not be alarmed if it crashes due to a GPU/Graphics failure on startup (that's because QEMU), but at least it should tell you if Pulsar successfully build

bryango commented:
Good news!

Result of `nixpkgs-review pr 292611 --extra-nixpkgs-config '{ allowInsecurePredicate = x: true; }'` run on `aarch64-linux`
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>pulsar</li>
  </ul>
</details>

Thank you for the help! I know how to build aarch64 now haha!

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/1520

pbsds commented on pkgs/applications/editors/pulsar/default.nix:
```diff
@@ -136,8 +136,6 @@ stdenv.mkDerivation rec {
       --set-rpath "${newLibpath}" \
       $opt/resources/app/ppm/bin/node
-    patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" \
-      $opt/resources/app.asar.unpacked/node_modules/symbols-view/vendor/ctags-linux
```
I see `opt/Pulsar/resources/app.asar.unpacked/node_modules/symbol-provider-ctags/vendor/ctags-linux` exists, should it be patchelf'ed?

bryango replied:
Thank you for the review!

> I see `opt/Pulsar/resources/app.asar.unpacked/node_modules/symbol-provider-ctags/vendor/ctags-linux` exists, should it be patchelf'ed?

Indeed I seem to have missed this... I think it would be better to still patch it at the new location; let me address this in a follow-up commit. It's easy for me to miss these kinds of issues as I am not running NixOS but nix as a package manager on an FHS, so the bug might be hidden from me haha

pbsds reviewed (commented):
Builds fine on my end.

bryango committed:
pulsar: 1.109.0 -> 1.114.0

Release notes: https://github.com/pulsar-edit/pulsar/releases/tag/v1.114.0

pulsar has been marked insecure so this change would not be verified by
ofborg. Still, it is very actively developed and quite enjoyable to use.
As an atom veteran, I would like to help keep it usable; thus adding
myself to the maintainers.

A patchelf fixup for `ctags-linux` is adjusted according to the new
location of the binary.

Co-authored-by: Peder Bergebakken Sundt <pbsds@hotmail.com>

pbsds commented:
Result of `nixpkgs-review pr 292611` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>pulsar</li>
  </ul>
</details>

Run fine, LGTM

pbsds commented:
Feel free to ping me on the next pulsar bump. It is my primary editor ;)

Thatoo commented:
Why does it say
`       error: Package ‘pulsar-1.114.0’ in /nix/store/n9hd1kf2kwn70x8i4vmqxk7xxxl93m6r-nixos-unstable/nixos-unstable/pkgs/applications/editors/pulsar/default.nix:202 is marked as insecure, refusing to evaluate.`
?
What does that mean? Will it ever remain insecure?
